### PR TITLE
add umap to benchmarking

### PR DESCRIPTION
**Pull request by rishic3**
also added some comments about dtype support and visualization differences to notebook, and minor max_records_per_batch change in umap.

### CODE REVIEW

File: python/src/spark_rapids_ml/umap.py [hunk 1]
```diff
@@ -343,7 +343,7 @@ def __init__(self, sample_fraction: float = 1.0, **kwargs: Any) -> None:
         super().__init__()
         self.set_params(**kwargs)
         self.sample_fraction = sample_fraction
-        self.maxRecordsPerBatch = 10000
+        self.max_records_per_batch = 10000
```
Comment (lijinf2) [inline python/src/spark_rapids_ml/umap.py:346]: Can we set this in the same way as line 369, i.e. through "spark.sql.execution.arrow.maxRecordsPerBatch"?

Comment (lijinf2) [inline python/src/spark_rapids_ml/umap.py:346]: Can we set this in the same way as line 370, i.e. through "spark.sql.execution.arrow.maxRecordsPerBatch"?

File: python/benchmark/benchmark/bench_umap.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            print(f"cpu total took: {total_time} sec")
+
+        report_dict = {
+            "fit": fit_time,
+            "transform": transform_time,
+            "total_time": total_time,
+            "num_gpus": num_gpus,
+            "num_cpus": num_cpus,
+            "no_cache": no_cache,
+            "train_path": self.args.train_path,
```
Comment (lijinf2) [inline python/benchmark/benchmark/bench_umap.py:164]: Is it possible to add one or two metrics to show the quality of umap and pca?

File: python/run_benchmark.sh [hunk 3]
```diff
@@ -390,3 +391,61 @@ if [[ "${MODE}" =~ "logistic_regression" ]] || [[ "${MODE}" == "all" ]]; then
...
+        --regParam 0.00001 \
+        --num_gpus $num_gpus \
+        --num_cpus $num_cpus \
+        --num_runs $num_runs \
+        --train_path "${gen_data_root}/classification/r${num_rows}_c${num_cols}_float32.parquet" \
+        --transform_path "${gen_data_root}/classification/r${num_rows}_c${num_cols}_float32.parquet" \
+        --report_path "report_logistic_regression_${cluster_type}.csv" \
+        $common_confs $spark_rapids_confs \
+        ${EXTRA_ARGS}
+fi
```
Comment (leewyang) [inline python/run_benchmark.sh:425]: Not sure why LogisticRegression benchmark shows as a change, does this PR need to be re-based now that the other PR is merged?

File: python/src/spark_rapids_ml/umap.py (hunk 1 above)
Comment (leewyang) [reply python/src/spark_rapids_ml/umap.py:346]: Basically, this line looks redundant w/ L369-373.  Not sure if that's intentional.

**Pull request closed without merging**